homeassistant/components/season/sensor.py

```diff
@@ -113,36 +118,20 @@ def get_season(date, hemisphere, season_tracking_type):
 class Season(SensorEntity):
     """Representation of the current season."""
 
-    def __init__(self, hass, hemisphere, season_tracking_type, name):
+    _attr_device_class = "season__season"
+
+    def __init__(self, hemisphere: str, season_tracking_type: str, name: str) -> None:
         """Initialize the season."""
-        self.hass = hass
-        self._name = name
+        self._attr_name = name
         self.hemisphere = hemisphere
-        self.datetime = None
         self.type = season_tracking_type
-        self.season = None
-
-    @property
-    def name(self):
-        """Return the name."""
-        return self._name
-
-    @property
-    def native_value(self):
-        """Return the current season."""
-        return self.season
 
-    @property
-    def device_class(self):
-        """Return the device class."""
-        return "season__season"
-
-    @property
-    def icon(self):
-        """Icon to use in the frontend, if any."""
-        return SEASON_ICONS.get(self.season, "mdi:cloud")
-
-    def update(self):
+    def update(self) -> None:
         """Update season."""
-        self.datetime = utcnow().replace(tzinfo=None)
-        self.season = get_season(self.datetime, self.hemisphere, self.type)
+        self._attr_native_value = get_season(
+            utcnow().replace(tzinfo=None), self.hemisphere, self.type
```

Comment:
This replicates the existing behaviour, so no new problem.

It might not be critical to know the precise start of a season to the nearest hour, but won't using utcnow() mean that the season transitions when the utc timezone enters the season rather than the local timezone?

Reply:
Yeah, that is actually an existing issue I would say